mod.ts

```diff
@@ -70,11 +89,37 @@ export interface Job {
     name: string;
     email: string;
   };
+}
+
+/**
+ * A job which was triggered by a webhook.
+ *
+ * Already contains ChangeSummary as this information is provided by the webhook.
+ */
+export interface WebhookJob extends BasicJob {
   /**
-   * Only used for full_update
+   * Relevant changes as provided by the webhook payload
    */
-  files?: {
-    modified?: string[];
-    removed?: string[];
-  };
+  files: ChangeSummary;
 }
+
+/**
+ * A job which was triggered by a request to /full_update
+ */
+export interface FullUpdateJob extends BasicJob {
+  /**
+   * Slice of files that were present when the full_update was triggered
+   */
+  files: { modified: string[] };
```

Comment:
I have introduced spearate types for the different jobs to make it clear which combinations of properties are ever present, instead of marking everthing as ` | undefined`. I have however opted to make the types overlap as much as possible, thus using a `files` object containing a single "`modifed`" array here so code can more easily handle the different types of job. (`files: { modified: string[] };` here is a subset of `files: ChangeSummary;` used in `WebhookJob`)